client_test.go

```diff
@@ -379,6 +385,83 @@ func TestClientReconnect(t *testing.T) {
 	}
 }
 
+func TestShimSockLength(t *testing.T) {
+	t.Parallel()
+
+	// Max length of namespace should be 76
+	namespace := strings.Repeat("n", 76)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ctx = namespaces.WithNamespace(ctx, namespace)
+	client, err := newClient(t, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if client == nil {
+		t.Fatal("New() returned nil client")
+	}
+	image, err := client.Pull(ctx, testImage,
+		WithPlatformMatcher(platforms.Default()),
+		WithPullUnpack,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	runtimeRoot, err := ioutil.TempDir("", "containerd-test-runtime-root")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(runtimeRoot)
+
+	for _, runtime := range []struct {
+		Name    string
+		Options interface{}
+	}{
+		{
+			plugin.RuntimeLinuxV1,
```

Comment:
I think we don't need to specify runtime for this case. The travis will run two jobs for v1 or v2 runtime by `TEST_RUNTIME`.